File: tasks/translation/filtered_dataset/codenet/Java/Code/s221078931_comments_removed.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

	public static void main(String[] args) throws IOException {


		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		IntegralRect rects[] = new IntegralRect[40000];
		for(int i = 0; i < 200; i++){
			for(int j = 0; j < 200; j++){
				rects[i*200 + j] = new IntegralRect(i + 1, j + 1);
			}
		}
		Arrays.sort(rects);
		









		while(true){
			String[] tmpArray = br.readLine().split(" ");

			int h = Integer.parseInt(tmpArray[0]);
			int w = Integer.parseInt(tmpArray[1]);

			if(h == 0 && w == 0){
				break;
			}

			IntegralRect tmpRect = new IntegralRect(h, w);
			int diag = tmpRect.diagonal();


			for(int i = 0; i < rects.length ; i++){


				if(rects[i].compareTo(tmpRect) > 0){
					System.out.println(rects[i].height+" "+rects[i].width);
					break;
				}
			}
		}
	}

}

class IntegralRect implements Comparable<IntegralRect> {
	int height;
	int width;
	public IntegralRect(int h, int w){
		this.height = h;
		this.width = w;
	}


	public int diagonal(){
		if(height >= width){
			return Integer.MAX_VALUE;
		}
		return this.height*this.height + this.width*this.width;
	}


	@Override
	public int compareTo(IntegralRect o) {


		int d1 = this.diagonal();
		int d2 = o.diagonal();


		return d1 == d2 ? this.height - o.height : d1 - d2;
	}
}

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C++/s221078931_comments_removed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <limits>

class IntegralRect {
public:
    int height;
    int width;

    IntegralRect(int h, int w) : height(h), width(w) {}

    int diagonal() const {
        if (height >= width) {
            return std::numeric_limits<int>::max();
        }
        return height * height + width * width;
    }

    int compareTo(const IntegralRect& other) const {
        int d1 = diagonal();
        int d2 = other.diagonal();
        if (d1 == d2) {
            return height - other.height;
        }
        return d1 - d2;
    }
};

bool compareRects(const IntegralRect& rect1, const IntegralRect& rect2) {
    return rect1.compareTo(rect2) < 0;
}

int main() {
    std::vector<IntegralRect> rects(40000);
    for (int i = 0; i < 200; i++) {
        for (int j = 0; j < 200; j++) {
            rects[i * 200 + j] = IntegralRect(i + 1, j + 1);
        }
    }
    std::sort(rects.begin(), rects.end(), compareRects);

    while (true) {
        int h, w;
        std::cin >> h >> w;
        if (h == 0 && w == 0) {
            break;
        }

        IntegralRect tmpRect(h, w);
        int diag = tmpRect.diagonal();

        for (const auto& rect : rects) {
            if (rect.compareTo(tmpRect) > 0) {
                std::cout << rect.height << " " << rect.width << std::endl;
                break;
            }
        }
    }
    // End of Code
}
